internal/dataplane/parser/ingressclassparemeters_test.go

```diff
@@ -137,21 +140,25 @@ func TestGetIngressClassParameters(t *testing.T) {
 				},
 			}
 			cacheStores, err := store.NewCacheStoresFromObjs()
-			assert.NoError(t, err)
+			require.NoError(t, err)
 			err = cacheStores.Add(ingressClass)
-			assert.NoError(t, err)
+			require.NoError(t, err)
 			err = cacheStores.Add(icp)
-			assert.NoError(t, err)
+			require.NoError(t, err)
 			s := store.New(cacheStores, ingressClass.Name, true, true, true, logrus.New())
 			icpSpec, err := getIngressClassParametersOrDefault(s)
 			assert.Truef(t, reflect.DeepEqual(*tc.parameterSpec, icpSpec),
 				fmt.Sprintf("should get same ingress parameter spec: expected %+v, actual %+v", tc.parameterSpec, icpSpec),
 			)
-			if tc.hasError {
-				assert.NotNil(t, err)
-				assert.Truef(t, errors.As(err, &store.ErrNotFound{}), "error should be a store.ErrNotFound, actual %T", err)
+
+			if tc.err != nil {
+				assert.EqualError(t, err, tc.err.Error())
+
+				if errors.As(err, &store.ErrNotFound{}) {
```

Comment:
```suggestion
				if errors.As(tc.err, &store.ErrNotFound{}) {
```
Shouldn't it be this way? 🤔